src/Streamly/Prelude.hs

```diff
@@ -367,6 +385,9 @@ notElem e m = go (toStream m)
 length :: (Streaming t, Monad m) => t m a -> m Int
 length = foldl (\n _ -> n + 1) 0 id
 
+reverse :: (Streaming t, Monad m) => t m a -> m (t m a)
```

Comment:
-- | Returns the elements of the stream in a reverse order. The stream must be finite.

The return type must be `t m a`.